### PR TITLE
process replay: prefix setup in helper

### PR DESCRIPTION
**Pull request by lukaspetersson**
prefix will be used in more places, move to helper file.

### CODE REVIEW

File: selfdrive/test/process_replay/helpers.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import os
+import shutil
+import uuid
+
+from common.params import Params
+
+def setup_prefix():
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:7]: might be nice to turn the whole thing into a context manager

Comment (lukaspetersson) [reply selfdrive/test/process_replay/helpers.py:7]: good idea!

File: selfdrive/test/process_replay/helpers.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import os
+import shutil
+import uuid
+
+from common.params import Params
+
+class Prefix(object):
+  def __init__(self, prefix=""):
+    self.prefix = prefix if prefix else str(uuid.uuid4())
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:9]: use None as the default

File: selfdrive/test/process_replay/helpers.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import os
+import shutil
+import uuid
+
+from common.params import Params
+
+class Prefix(object):
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:7]: call this OpenpilotPrefix

File: selfdrive/test/process_replay/helpers.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  def __enter__(self):
+    os.environ['OPENPILOT_PREFIX'] = self.prefix
+    try:
+      os.mkdir(self.msgq_path)
+    except FileExistsError:
+      pass
+
+  def __exit__(self, exc_type, exc_obj, exc_tb):
+    if not os.environ.get("OPENPILOT_PREFIX", 0):
+      return
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:21]: now that the prefix is in the class, think it's preferable to skip this

File: selfdrive/test/process_replay/helpers.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import os
+import shutil
+import uuid
+
+from common.params import Params
+
+class OpenpilotPrefix(object):
+  def __init__(self, prefix=None):
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:8]: add type hints

File: selfdrive/test/process_replay/helpers.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    self.msgq_path = os.path.join('/dev/shm', self.prefix)
+
+  def __enter__(self):
+    os.environ['OPENPILOT_PREFIX'] = self.prefix
+    try:
+      os.mkdir(self.msgq_path)
+    except FileExistsError:
+      pass
+
+  def __exit__(self, exc_type, exc_obj, exc_tb):
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/helpers.py:19]: also unset OPENPILOT_PREFIX in the environment

**Comment by adeebshihadeh** [comment]
LGTM, but can you also set the jobs flag (use nproc) in CI to make sure we're testing parallel process replay in CI?

**Comment by lukaspetersson** [comment]
> LGTM, but can you also set the jobs flag (use nproc) in CI to make sure we're testing parallel process replay in CI?

Ready to merge then?
made a separate PR for this: https://github.com/commaai/openpilot/pull/24646

**Pull request merged**